x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/DataType.java

```diff
@@ -287,6 +286,10 @@ public enum DataType {
         map.put("point", DataType.CARTESIAN_POINT);
         map.put("shape", DataType.CARTESIAN_SHAPE);
         ES_TO_TYPE = Collections.unmodifiableMap(map);
+        // DATETIME has different esType and typeName, add an entry in NAME_TO_TYPE with date as key
+        map = TYPES.stream().collect(toMap(DataType::typeName, t -> t));
+        map.put("date", DataType.DATETIME);
```

Comment:
I looked into `NAME_OR_ALIAS_TO_TYPE` a bit further, and it is used by `inline_cast` only for now. It should have `date` I think, so that we can support `::date`, today only `::datetime` is supported. I will open a separate issue to do it, as it requires additional tests, and I'd like to address it separately.